pkg/util/retry/retrier_test.go

```diff
@@ -314,3 +314,29 @@ func TestBackoff(t *testing.T) {
 	expectedNext = now.Add(maxRetryDelay)
 	assert.WithinDuration(t, expectedNext, mocked.NextRetry(), time.Millisecond)
 }
+
+func TestLastError(t *testing.T) {
+	mocked := &DummyLogic{}
+	mocked.On("Attempt").Return(errors.New("nope")).Once()
+	config := &Config{
+		Name:          "mocked",
+		AttemptMethod: mocked.Attempt,
+		Strategy:      RetryCount,
+		RetryCount:    5,
+		RetryDelay:    100 * time.Millisecond,
+	}
+	err := mocked.SetupRetrier(config)
+	assert.Nil(t, err)
+
+	// First call will trigger an attempt
+	_ = mocked.TriggerRetry()
+	err = mocked.LastError()
+	assert.True(t, IsErrWillRetry(err))
+	time.Sleep(time.Nanosecond) // Make sure we expire the delay
```

Comment:
Shouldn't this be at least `config.RetryDelay` to be sure?

Reply:
indeed, done!